src/views/TradingCompetition/components/BattleCta/index.tsx

```diff
@@ -0,0 +1,175 @@
+
+const StyledButton = styled(Button)`
+  margin: 16px 20px 0;
+  z-index: 200;
+
+  svg {
+    margin: 0 4px 0 0;
+    height: 20px;
+    width: auto;
+    fill: ${({ theme }) => theme.colors.textDisabled};
+  }
+`
+
+const BattleCta: React.FC<CompetitionProps> = ({
+  userTradingStats,
+  account,
+  isCompetitionLive,
+  profile,
+  isLoading,
+  hasCompetitionFinished,
+  userCanClaim,
+  userRewards,
+  onRegisterSuccess,
+}) => {
+  const TranslateString = useI18n()
+  const { login, logout } = useAuth()
+  const { onPresentConnectModal } = useWalletModal(login, logout)
+  const [onPresentRegisterModal] = useModal(
+    <RegisterModal profile={profile} onRegisterSuccess={onRegisterSuccess} />,
+    false,
+  )
+  const [onPresentClaimModal] = useModal(<ClaimModal />, false)
+
+  const { hasRegistered, hasClaimed } = userTradingStats
+
+  const getHeadingText = () => {
+    // Competition live
+    if (isCompetitionLive) {
+      return TranslateString(999, 'Now Live!')
+    }
+    // Competition finished
+    if (hasCompetitionFinished) {
+      return TranslateString(999, 'Finished!')
+    }
+    // Competition not started
+    return TranslateString(999, 'Starting Soon')
+  }
+
+  const getButtonText = () => {
+    // No wallet connected
+    if (!account) {
+      return TranslateString(999, 'Connect Wallet')
+    }
+
+    // User not registered
+    if (!hasRegistered) {
+      return TranslateString(999, 'Register Now!')
+    }
+
+    if (hasRegistered) {
+      // User registered and competition live
+      if (isCompetitionLive) {
+        return TranslateString(999, 'Trade Now')
+      }
+
+      // User registered and competition finished
+      if (hasCompetitionFinished) {
+        // User has prizes to claim
+        if (userCanClaim) {
+          return TranslateString(999, 'Claim prizes')
+        }
+        // User has already claimed prizes
+        if (hasClaimed) {
+          return (
+            <>
+              <CheckmarkCircleIcon /> {TranslateString(999, 'Prizes Claimed!')}
+            </>
+          )
+        }
+        // Use has nothing to claim
+        if (!userCanClaim) {
+          return TranslateString(999, 'Nothing to claim')
+        }
+      }
+
+      // User registered but competition has not started
+      if (!isCompetitionLive) {
+        return (
+          <>
+            <CheckmarkCircleIcon /> {TranslateString(999, 'Registered!')}
+          </>
+        )
+      }
+    }
+
+    // May be useful for debugging - if somehow none of the above conditions are met
+    return 'Whoopsie'
+  }
+
+  const handleCtaClick = () => {
+    // No wallet connected
+    if (!account) {
+      onPresentConnectModal()
+    }
+    // Wallet connected but user not registered
+    if (account && !hasRegistered) {
+      onPresentRegisterModal()
+    }
+    // Registered and competition is live
+    if (hasRegistered && isCompetitionLive) {
+      window.location.href = 'https://exchange.pancakeswap.finance/#/swap'
```

Comment:
I think the history object is a bit problematic here because it routes relative to the domain? I need to send the user to the exchange domain.

so :
```js
history.push('https://exchange.pancakeswap.finance/#/swap')
```

sends you to (with localhost:3000 as the example):
```
http://localhost:3000/https://exchange.pancakeswap.finance/#/swap
```